### PR TITLE
Improve output formatting for mismatched versions

### PR DESCRIPTION
**Pull request by ddzz**
Before:

Screen Shot 2021-09-05 at 2 35 20 PM

After:

Screen Shot 2021-09-05 at 2 55 24 PM

### CODE REVIEW

File: lib/output.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import chalk from 'chalk';
+
```
Comment (bmish) [inline lib/output.ts:2]: Remove extra line:
```suggestion
```

File: lib/output.ts [hunk 2]
```diff
@@ -6,15 +8,17 @@ export function mismatchingVersionsToOutputLines(
...
+      )} has more than one version:\n  ${obj.versions
         .sort((a, b) => compareRanges(a.version, b.version))
         .map(
           (versionObj) =>
             `${versionObj.version} (${versionObj.count} ${pluralizeUsage(
               versionObj.count
             )})`
         )
-        .join(', ')}`
+        .join('\n  ')}`
```
Comment (bmish) [inline lib/output.ts:21]: We should only join with a newline `\n` here and no spaces. Inside `.map(` above, we can add a tab character `\t` to indent each line.

**Review by bmish — CHANGES_REQUESTED** [review]

File: lib/output.ts [hunk 3]
```diff
@@ -6,15 +8,17 @@ export function mismatchingVersionsToOutputLines(
...
     (obj) =>
-      `${obj.dependency} has more than one version: ${obj.versions
+      `${chalk.bold(
+        obj.dependency
+      )} has more than one version:\n  ${obj.versions
         .sort((a, b) => compareRanges(a.version, b.version))
         .map(
           (versionObj) =>
-            `${versionObj.version} (${versionObj.count} ${pluralizeUsage(
+            `${chalk.greenBright(versionObj.version)} (${versionObj.count} ${pluralizeUsage(
```
Comment (bmish) [inline lib/output.ts:17]: Why green? Need a good reason to use colors.

File: lib/output.ts [hunk 4]
```diff
@@ -6,15 +7,18 @@ export function mismatchingVersionsToOutputLines(
...
     (obj) =>
-      `${obj.dependency} has more than one version: ${obj.versions
+      `${chalk.bold(
+        obj.dependency
+      )} has more than one version:\n  ${obj.versions
         .sort((a, b) => compareRanges(a.version, b.version))
         .map(
           (versionObj) =>
-            `${versionObj.version} (${versionObj.count} ${pluralizeUsage(
+            // eslint-disable-next-line prettier/prettier
```
Comment (bmish) [inline lib/output.ts:16]: Prefer not to disable prettier.

File: lib/output.ts (hunk 3 above)
Comment (ddzz) [reply lib/output.ts:17]: It makes it easier to read. I used a color that's similar to what we see in test output.

Comment (bmish) [reply lib/output.ts:17]: Green indicates success. I don't think it makes sense here.

Comment (ddzz) [reply lib/output.ts:17]: Good point, I made it red.

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "check-dependency-version-consistency",
-  "version": "1.1.0",
+  "version": "1.1.1",
```
Comment (bmish) [inline package.json:3]: I have an automated release process with `yarn release` that will be done separately from PRs.

**Comment by bmish** [comment]
Might be missing yarn.lock from changes.

**Comment by bmish** [comment]
README.md example needs to be updated with the new format.

**Comment by ddzz** [comment]
> README.md example needs to be updated with the new format.

Fixed

**Comment by bmish** [comment]
I'm guessing the tests are failing because the github actions terminal has color output turned off: https://github.com/chalk/chalk#supportscolor

**Comment by ddzz** [comment]
> I'm guessing the tests are failing because the github actions terminal has color output turned off: https://github.com/chalk/chalk#supportscolor

That's what I suspected. Fixed it by adding the `--colors` flag to the test script.

**Review by bmish — APPROVED** [review]
Nice work!

**Pull request merged**